dispatches/models/nuclear_case/chgtank.py

```diff
@@ -0,0 +1,590 @@
+                self.flowsheet().config.time.first()]
+                .get_material_flow_basis() == MaterialFlowBasis.molar):
+            flow_units = units("flow_mole")
+        elif (self.control_volume.properties_in[
+                self.flowsheet().config.time.first()]
+              .get_material_flow_basis() == MaterialFlowBasis.mass):
+            flow_units = units("flow_mass")
+        else:
+            flow_units = None
+
+        # compute gas constant value in the base units
+        gas_constant = Ideal.gas_constant(
+            self.control_volume.properties_in[0])
+
+        # Add Inlet and Outlet Ports
+        self.add_inlet_port()
+        self.add_outlet_port()
+
+        # Define Vars for Tank volume calculations
+        self.tank_diameter = Var(
+            self.flowsheet().config.time,
+            within=NonNegativeReals,
+            initialize=1.0,
+            doc="Diameter of storage tank in m",
+            units=pyunits.m)
+
+        self.tank_length = Var(
+            self.flowsheet().config.time,
+            within=NonNegativeReals,
+            initialize=1.0,
+            doc="Length of storage tank in m",
+            units=pyunits.m)
+
+        # Tank volume calculation
+        @self.Constraint(self.flowsheet().config.time)
+        def volume_cons(b, t):
+            return (b.control_volume.volume[t] ==
+                    const.pi * b.tank_length[t] *
+                    ((b.tank_diameter[t]/2)**2))
+
+        # define Vars for the model
+        self.dt = Var(
+            self.flowsheet().config.time,
+            domain=NonNegativeReals,
+            initialize=100,
+            doc="time step for holdup calculation",
+            units=time_units)
+
+        self.heat_duty = Var(
+            self.flowsheet().config.time,
+            domain=Reals,
+            initialize=0.0,
+            doc="Heat transfered through walls, is 0 for adiabatic",
+            units=power_units)
+
+        self.material_accumulation = Var(
+            self.flowsheet().config.time,
+            pc_set,
+            within=Reals,
+            initialize=1.0,
+            doc="Accumulation of material in tank",
+            units=flow_units)
+
+        self.energy_accumulation = Var(
+            self.flowsheet().config.time,
+            phase_list,
+            within=Reals,
+            initialize=1.0,
+            doc="Energy accumulation",
+            units=power_units)
+
+        self.material_holdup = Var(
+            self.flowsheet().config.time,
+            pc_set,
+            within=Reals,
+            initialize=1.0,
+            doc="Material holdup in tank",
+            units=flow_units*time_units)
+
+        self.energy_holdup = Var(
+            self.flowsheet().config.time,
+            phase_list,
+            within=Reals,
+            initialize=1.0,
+            doc="Energy holdup in tank",
+            units=energy_units)
+
+        self.initial_material_holdup = Var(
```

Comment:
"initial" seems like a poor choice of name here if this is indexed by time - it is actually the holdup at the previous time step. "initial" implies it only applies at t==0.